stdlib/public/core/Indices.swift

```diff
@@ -115,8 +84,23 @@ public struct ${Self}<
   internal var _endIndex: Elements.Index
 }
 
-extension ${collectionForTraversal(Traversal)}
-where Indices == ${Self}<Self> {
+extension DefaultIndices: BidirectionalCollection where Elements: BidirectionalCollection {
+  @_inlineable
+  public func index(before i: Index) -> Index {
+    // FIXME: swift-3-indexing-model: range check.
+    return _elements.index(before: i)
+  }
+
+  @_inlineable
+  public func formIndex(before i: inout Index) {
+    // FIXME: swift-3-indexing-model: range check.
+    _elements.formIndex(before: &i)
+  }
+}
+
+extension DefaultIndices: RandomAccessCollection where Elements: RandomAccessCollection { }
```

Comment:
Need to make sure we're forwarding distance(to:) and index(_:offsetBy:) to the underlying collection.